test/functional/test-viewer.js

```diff
@@ -211,4 +213,139 @@ describe('Viewer', () => {
     expect(delivered[1].eventType).to.equal('documentResized');
     expect(delivered[1].data.width).to.equal(13);
   });
+
+  describe('isTrustedViewer', () => {
+    function test(referrer, toBeTrusted) {
+      windowApi.parent = {};
+
+      windowApi.location.ancestorOrigins = referrer ? [referrer] : [];
+      expect(new Viewer(windowApi).isTrustedViewer())
+          .to.equal(toBeTrusted, 'by ancestor');
+      windowApi.location.ancestorOrigins = null;
+
+      windowApi.document.referrer = referrer;
+      expect(new Viewer(windowApi).isTrustedViewer())
+          .to.equal(toBeTrusted, 'by referrer');
+      windowApi.document.referrer = '';
+    }
+
+    it('should verify ancestorOrigins ahead of referrer', () => {
+      windowApi.parent = {};
+      windowApi.document.referrer = 'https://google.com';
+      expect(new Viewer(windowApi).isTrustedViewer()).to.equal(true);
+
+      windowApi.location.ancestorOrigins = ['https://google.com'];
+      expect(new Viewer(windowApi).isTrustedViewer()).to.equal(true);
+
+      windowApi.location.ancestorOrigins = ['https://google.other.com'];
+      expect(new Viewer(windowApi).isTrustedViewer()).to.equal(false);
+
+      // Even empty ancestorOrigins takes precendence.
+      windowApi.location.ancestorOrigins = [];
+      expect(new Viewer(windowApi).isTrustedViewer()).to.equal(false);
+    });
+
+    it('should only consider viewer trusted when iframed', () => {
+      windowApi.parent = {};
+      windowApi.location.ancestorOrigins = ['https://google.com'];
+      expect(new Viewer(windowApi).isTrustedViewer()).to.equal(true);
+
+      windowApi.parent = windowApi;
+      expect(new Viewer(windowApi).isTrustedViewer()).to.equal(false);
+    });
+
+    it('should flag viewer as not trusted without referrer or ancestor', () => {
+      test('', false);
+    });
+
+    it('should trust host as referrer with https', () => {
+      test('https://google.com', true);
+    });
+
+    it('should not trust host as referrer with http', () => {
+      test('http://google.com', false);
+    });
+
+    it('should trust domain variations', () => {
+      test('https://google.com', true);
+      test('https://www.google.com', true);
+      test('https://news.google.com', true);
+      test('https://www.google.co.uk', true);
+      test('https://www.google.co.au', true);
+      test('https://news.google.co.uk', true);
+      test('https://news.google.co.au', true);
+    });
+
+    it('should NOT trust wrong or non-whitelisted domain variations', () => {
+      test('https://google.other.com', false);
+      test('https://google.net', false);
```

Comment:
Please see request for more test cases I sent.

Would also be good to have more good cases.